src/app/utils/useSet.tsx

```diff
@@ -1,59 +1,35 @@
 import { useEffect, useState } from "react";
 
-type SearchResultProps = {
-  set_num: string;
-  name: string;
-  num_parts: number;
-  set_img_url: string;
-  theme_id: number;
-  detail: string | undefined;
+export type SetProps = {
```

Comment:
It's unusual to export props. You could instead create a separate Type `type Set` (for example in a `types` folder) and reuse that type.